tests/manage/z_cluster/cluster_expansion/test_add_capacity_with_node_restart.py

```diff
@@ -0,0 +1,169 @@
+# When this branch will be merged into master I will use the functions from this branch.
+
+def get_percent_used_capacity():
+    """
+    Function to calculate the percentage of used capacity in a cluster
+
+    Returns:
+        float: The percentage of the used capacity in the cluster
+    """
+    ct_pod = pod_helpers.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd='ceph df')
+    total_used = (output.get('stats').get('total_used_raw_bytes'))
+    total_avail = (output.get('stats').get('total_bytes'))
+    return 100.0 * total_used / total_avail
+
+
+def check_pods_in_running_state(namespace=defaults.ROOK_CLUSTER_NAMESPACE):
+    """
+    checks whether all the pods in a given namespace are in Running state or not
+    Returns:
+        Boolean: True, if all pods in Running state. False, otherwise
+    """
+    ret_val = True
+    list_of_pods = pod_helpers.get_all_pods(namespace)
+    ocp_pod_obj = OCP(kind=constants.POD, namespace=namespace)
+    for p in list_of_pods:
+        # we don't want to compare osd-prepare and canary pods as they get created freshly when an osd need to be added.
+        if "rook-ceph-osd-prepare" not in p.name and "rook-ceph-drain-canary" not in p.name:
+            status = ocp_pod_obj.get_resource(p.name, 'STATUS')
+            if status not in "Running":
+                logging.error(f"The pod {p.name} is in {status} state. Expected = Running")
+                ret_val = False
+    return ret_val
+
+
+@pytest.mark.parametrize(
+    argnames=["node_type", "num_of_nodes"],
+    argvalues=[
+        pytest.param(
+            *['worker', 1],
+            marks=pytest.mark.polarion_id("OCS-1313")
+        ),
+     ]
+)
+@pytest.mark.parametrize(
+    "workload_storageutilization_rbd",
+    [(0.25, False, 120)],
+    indirect=["workload_storageutilization_rbd"]
+)
+# @pytest.mark.parametrize(
+#     "workload_storageutilization_cephfs",
+#     [(0.05, False, 120)],
+#     indirect=["workload_storageutilization_cephfs"]
+# )
+@ignore_leftovers
+@tier4
+@polarion_id('OCS-1313')
+class TestAddCapacityNodeRestart(ManageTest):
+    """
+    Test add capacity when one of the nodes got restart
+    in the middle of the process. Don't forget to include the flag '--cluster-name' when running the test
+    """
+    num_of_pvcs = 3
+    pvc_size = 100
+
+    def test_add_capacity_node_restart(
+        self, nodes, multi_pvc_factory, pvc_factory, pod_factory, workload_storageutilization_rbd,
+        node_type, num_of_nodes,
+    ):
+        """ test add capacity when one of the nodes got restart
+        in the middle of the process
+        """
+        logging.info("Condition 1 to start the test is met: storageutilization is completed")
+        used_capacity_after_utilizing = get_percent_used_capacity()
+        logging.info(f"### used capacity after storageutilization = {used_capacity_after_utilizing}")
+
+        seconds_to_wait_after_storageutilization = 60
```

Comment:
please don't wait for an absolute number of seconds. Instead create a functionality to poll what you need.
You can use `ocs_ci.utility.utils.TimeoutSampler` or ocs_ci.utility.retry.retry